### PR TITLE
Add skipif managed service marker and update it in OSD skip tests

### PR DESCRIPTION
**Pull request by suchita-g**
Created a common skipif marker for rosa and OSD
The test skip applicable for OSD platform is also
applicable for rosa platform. hence replacing
skipif_openshiftdedicated marker with skipif_managed_service
Signed-off-by: suchita.gatfane <sgatfane@redhat.com>

### CODE REVIEW

**Comment by suchita-g** [comment]
Verification Job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/9670/
skipped result is as expected :    http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/sgatfane-rc130/sgatfane-rc130_20220130T184121/logs/test_report_1643914165.html

File: tests/manage/storageclass/conftest.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from ocs_ci.framework import config
+from ocs_ci.ocs.constants import MANAGED_SERVICE_PLATFORMS
+
+log = logging.getLogger(__name__)
+
+
+def pytest_collection_modifyitems(items):
+    """
+    A pytest hook to filter out storageclass tests
+    when running on openshift dedicated platform
```
Comment (jilju) [inline tests/manage/storageclass/conftest.py:11]:
```suggestion
    when running on managed service platforms
```

File: tests/manage/storageclass/conftest.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    Args:
+        items: list of collected tests
+
+    """
+    # Need to update the condition when MCG will get supported
```
Comment (jilju) [inline tests/manage/storageclass/conftest.py:17]: This statement is not appropriate for storage class tests. Please modify the statement.

**Review by jilju — APPROVED** [review]

**Review by sidhant-agrawal — APPROVED** [review]

**Review by mbukatov — APPROVED** [review]
Ack from blue team.

**Review by petr-balogh — APPROVED** [review]

**Review by prsurve — APPROVED** [review]

**Review by MeridianExplorer — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**